lumispy/signals/luminescence_spectrum.py

```diff
@@ -562,6 +586,49 @@ def px_to_nm_grating_solver(
         "\n", "\n\t"
     )
 
+    def centroid(self, slice=None, **kwargs):
+        """
+        Finds the centroid (center of mass) of a peak in the spectrum from the wavelength (or pixel number) and the intensity at each pixel value. It basically represents a "weighted average" of the peak.
+
+        Notes
+        -------
+        This function only works for a single peak. If you have multiple peaks, slice the signal beforehand or use the slice parameter.
+
+        TODO: Implement this function for multiple peaks (with the npeaks parameter) by finding the top 2 peaks from mean spectrum and then returning a signal with 2 com.
+
+        Parameters
+        -------
+        slice : tuple (2)
```

Comment:
In HyperSpy this parameter is in fact called `signal_range`, see e.g. [remove_background](https://hyperspy.readthedocs.io/en/latest/api/hyperspy._signals.signal1d.html#hyperspy._signals.signal1d.Signal1D.remove_background), and we should be consistent there. HyperSpy allows for "interactive" slicing, which would be a worthwile addition.

Reply:
I have changed the name of the parameter to match the standards.
However, adding an "interactive" functionality would require a significant amount of extra coding I don't think would be worth it. If you still think it is necessary, we can delay this PR more and I can attempt to add it in the future. I'd simply leave it as is and maybe add it in the future as an additional feature.

Reply:
Yes, lets keep it as a potential future feature. Just noticed that similar HyperSpy functions have that option - though I personally do not tend to use it much, I know some other people might appreciate.